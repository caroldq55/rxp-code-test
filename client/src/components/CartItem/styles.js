import styled from "styled-components";

const CartItemContainer = styled.div`
  margin: 5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const ItemName = styled.span`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;

export { CartItemContainer, ItemName };
