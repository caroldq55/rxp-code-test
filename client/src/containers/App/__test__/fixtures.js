import { fromJS } from 'immutable';

const products = [
  {
    isPublished: 'true',
    productName: 'Apple iPhone X',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-x/iphone-x-silver-grid.png',
    price: '299'
  },
  {
    isPublished: 'true',
    productName: 'Apple iPhone 8',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8-silver-grid.png',
    price: '100'
  },
  {
    isPublished: 'false',
    productName: 'Apple iPhone 8 Plus',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8plus-space-grey-grid.png',
    price: '99'
  },
  {
    isPublished: 'true',
    productName: 'Sony Xperia XA2',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
    price: '19'
  }
];

const item = {
  isPublished: true,
  productName: 'Sony Xperia XA2',
  productImage:
    'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
  price: '19'
};

const cartItems = [
  {
    isPublished: true,
    productName: 'Sony Xperia XA2',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
    price: '19'
  }
];

const publishedProducts = [
  {
    isPublished: true,
    productName: 'Apple iPhone X',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-x/iphone-x-silver-grid.png',
    price: '299'
  },
  {
    isPublished: true,
    productName: 'Apple iPhone 8',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8-silver-grid.png',
    price: '100'
  },
  {
    isPublished: true,
    productName: 'Sony Xperia XA2',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
    price: '19'
  }
];

const availableProducts = [
  {
    isPublished: true,
    productName: 'Apple iPhone X',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-x/iphone-x-silver-grid.png',
    price: '299'
  },
  {
    isPublished: true,
    productName: 'Apple iPhone 8',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8-silver-grid.png',
    price: '100'
  }
];

const initialValue = fromJS({
  products: [],
  loading: false,
  cartItems: []
});

export {
  products,
  initialValue,
  cartItems,
  availableProducts,
  item,
  publishedProducts
};
