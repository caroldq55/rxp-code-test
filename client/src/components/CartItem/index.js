import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

import { CartItemContainer, ItemName } from './styles';

const CartItem = ({ item, handleRemoveCartItem }) => (
  <CartItemContainer>
    <ItemName>{item.productName}</ItemName>
    <Button
      variant='warning'
      onClick={() => {
        handleRemoveCartItem(item);
      }}
      data-test-id='removeButton'
    >
      Remove
    </Button>
  </CartItemContainer>
);

CartItem.propTypes = {
  item: PropTypes.shape({
    productImage: PropTypes.string.isRequired,
    productName: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    isPublished: PropTypes.bool
  }).isRequired,
  handleRemoveCartItem: PropTypes.func.isRequired
};

export default CartItem;
