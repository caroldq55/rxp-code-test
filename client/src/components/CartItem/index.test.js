import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";

import CartItem from ".";
import { item } from "../../containers/App/__test__/fixtures";

describe("<CartItem/>", () => {
  const handleRemoveCartItemSpy = jest.fn();

  it("should render CartItem correctly", () => {
    const wrapper = shallow(
      <CartItem item={item} handleRemoveCartItem={handleRemoveCartItemSpy} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render CartItem correctly", () => {
    const wrapper = shallow(
      <CartItem item={item} handleRemoveCartItem={handleRemoveCartItemSpy} />
    );
    const removeButton = wrapper.find('[data-test-id="removeButton"]');
    expect(removeButton).toHaveLength(1);
    removeButton.simulate("click");
    expect(handleRemoveCartItemSpy).toHaveBeenCalledTimes(1);
  });
});
