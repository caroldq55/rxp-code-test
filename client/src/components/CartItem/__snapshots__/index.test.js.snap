// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem/> should render CartItem correctly 1`] = `ShallowWrapper {}`;
